### PR TITLE
Change duplicates error level

### LINKED ISSUE #178: Review error levels for duplicates
From the GTFS specification, ["Field types" section] for `ID`.

> An ID field value is an internal ID, not intended to be shown to riders, and is a sequence of any UTF-8 characters. Using only printable ASCII characters is recommended. **An ID is labeled "unique ID" when it must be unique within a file.**

When the type is marked as `Unique ID` I think it should be an `ERROR` severity instead of a `WARNING`.

The relevant code path seems to be here.

https://github.com/etalab/transport-validator/blob/0d120e25e68a4741d156f085f7f04affe51810a7/src/validators/raw_gtfs.rs#L23-L40

a duplicate ID for `stops.id` should be an `ERROR`. Other fields should be reviewed, we could check for more duplicates as well.

### PR DESCRIPTION
**Pull request by antoine-de**
Change it for all checks from warning to errors.

Also add checks for pathways, agencies and shapes

fixes #178

### CODE REVIEW

**Review by Tristramg — APPROVED** [review]
Indeed duplicates are really errors that will result in impossible situations.

**Comment by AntoineAugusti** [comment]
Wonderful, thanks @antoine-de!

File: readme.md [hunk 1]
```diff
@@ -123,7 +123,7 @@ Here is a human friendly list of them :
 | NullDuration | Warning | The travel duration between two stops is null. |
 | MissingLanguage | Warning | The publisher language code is missing. |
 | InvalidLanguage | Warning | The publisher language code is not valid. |
-| DuplicateObjectId | Warning | The object has at least one object with the same id. |
+| DuplicateObjectId | Error | The object has at least one object with the same id. |
```
Comment (AntoineAugusti) [inline readme.md]:
```suggestion
| DuplicateObjectId | Error | The object has at least one object with the same ID. |
```

**Review by AntoineAugusti — APPROVED** [review]

**Comment by AntoineAugusti** [comment]
Feel free to merge when you want

File: src/validators/raw_gtfs.rs [hunk 2]
```diff
@@ -20,23 +19,16 @@ fn check_duplicates<O: gtfs_structures::Id + gtfs_structures::Type>(
...
-    check_duplicates(&raw_gtfs.stops, Severity::Warning)
+    check_duplicates(&raw_gtfs.stops)
         .into_iter()
-        .chain(check_duplicates(&raw_gtfs.routes, Severity::Warning).into_iter())
-        .chain(check_duplicates(&raw_gtfs.trips, Severity::Warning).into_iter())
+        .chain(check_duplicates(&raw_gtfs.routes).into_iter())
+        .chain(check_duplicates(&raw_gtfs.trips).into_iter())
+        .chain(check_duplicates(&raw_gtfs.agencies).into_iter())
+        .chain(check_duplicates(&raw_gtfs.pathways.as_ref().unwrap_or(&Ok(vec![]))).into_iter())
+        .chain(check_duplicates(&raw_gtfs.shapes.as_ref().unwrap_or(&Ok(vec![]))).into_iter())
```
Comment (AntoineAugusti) [inline src/validators/raw_gtfs.rs:28]: Primary key for shapes should be `(shape_id, shape_pt_sequence)` https://gtfs.org/schedule/reference/#shapestxt, sorry I saw it too late!

**Pull request merged**